src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -106,7 +106,11 @@ public override string GetMessage(IFormatProvider? formatProvider = null)
 
                 public override bool Equals(object? obj)
                 {
-                    return _original.Equals(obj);
+                    if (obj instanceof CompilerDiagnostic cd) {
```

Comment:
I think you've been doing a bit too much Java 😉 
```suggestion
                    if (obj is CompilerDiagnostic cd) {
```

Reply:
Thanks.  Was hoping I could get away with not setting up this repo as an environment.